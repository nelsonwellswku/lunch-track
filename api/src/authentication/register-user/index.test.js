jest.mock('bcrypt', () => ({ hash: jest.fn().mockImplementation(() => Promise.resolve('testHashedPassword')) }));

const mockQueries = {
  doesUserExist: jest.fn(),
  createAppUser: jest.fn(),
};
jest.mock('./queries', () => mockQueries);

const registerUser = require('./index');

describe('register user endpoint;', () => {
  test('can register a new user', async () => {
    const req = {
      body: {
        emailAddress: 'TestUsername',
        password: 'TestPassword',
      },
    };

    const res = {
      sendStatus: jest.fn(),
    };

    await registerUser(req, res);

    expect(mockQueries.createAppUser).toMatchSnapshot('create app user');
    expect(res.sendStatus).toMatchSnapshot('send status');
  });
});
