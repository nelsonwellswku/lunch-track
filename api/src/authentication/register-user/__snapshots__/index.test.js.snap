// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register user endpoint; can register a new user: create app user 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "emailAddress": "TestUsername",
        "passwordHash": "testHashedPassword",
      },
    ],
  ],
}
`;

exports[`register user endpoint; can register a new user: send status 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      201,
    ],
  ],
}
`;
